File: alibreo/JGV/src/main/java/Movie.java
import java.time.Duration;

public class Movie {
    private final String title;
    private final Duration runningTime;
    private final Money fee;
    private final DiscountPolicy discountPolicy;

    public Movie(String title, Duration runningTime, Money fee, DiscountPolicy discountPolicy) {
        this.title = title;
        this.runningTime = runningTime;
        this.fee = fee;
        this.discountPolicy = discountPolicy;
    }

    public Money calculateMovieFee(Screening screening) {
        // 이건 생성자가 맡아야할 책임같지만..? -> 아니네
        if (discountPolicy == null) {
            return fee;
        }

        return fee.minus(discountPolicy.calculateDiscountAmount(screening));
    }

    public Money getFee() {
        return fee;
    }
}

File: prravda/src/theater-booking-application/movie.ts
import { Screening } from "./screening";
import { AbstractDiscountPolicy } from "./discount-policies/abstract.discount-policy";
import { Money } from "./money";

export class Movie {
  constructor(
    private title: string,
    private runningTime: Duration,
    private fee: Money,
    private discountPolicy: AbstractDiscountPolicy
  ) {}

  public getFee() {
    return this.fee;
  }

  public calculateMovieFee(screening: Screening) {
    return this.fee.minus(
      this.discountPolicy.calculateDiscountAmount(screening)
    );
  }
}
